### PR TITLE
Some refactorings in Stack.Build.Haddock

### PR DESCRIPTION
**Pull request by sjakobi**
Among other things:

* Fix a comment that breaks haddocks generation for stack

* Rename the `generate*HaddockIndex` functions to `generate*Haddocks` as they generate both the index as well as the actual haddocks

### CODE REVIEW

**Comment by borsboom** [comment]
I don't think `generate*HaddockIndex` should be renamed, as it actually doesn't generate any of the haddocks themselves, it only creates the index and contents and puts dependencies' haddocks in the right place.  Stack.Build.Execute.singleBuild is where the haddocks are generated.

**Comment by sjakobi** [comment]
Thanks for clearing that up, @borsboom!

File: src/Stack/Build/Haddock.hs [hunk 1]
```diff
@@ -209,16 +208,17 @@ generateHaddockIndex descr envOverride wc dumpPackages docRelFP destDir = do
                                 , srcInterfaceModTime
                                 , srcInterfaceAbsFile
                                 , destInterfaceAbsFile )
+    tryGetModificationTime :: FilePath -> IO (Either () UTCTime)
+    tryGetModificationTime = tryJust (guard . isDoesNotExistError) . getModificationTime
+    copyPkgDocs :: (a, UTCTime, Path Abs File, Path Abs File) -> IO ()
     copyPkgDocs (_,srcInterfaceModTime,srcInterfaceAbsFile,destInterfaceAbsFile) = do
-        -- | Copy dependencies' haddocks to documentation directory.  This way, relative @../$pkg-$ver@
+        -- Copy dependencies' haddocks to documentation directory.  This way, relative @../$pkg-$ver@
```
Comment (sjakobi) [inline src/Stack/Build/Haddock.hs]: In order to avoid problems like this one, would it be a good idea to let Travis build the haddocks too?

Comment (borsboom) [inline src/Stack/Build/Haddock.hs]: Haddocks are built as part of the integration tests, but those are run less frequently to try to keep a decent turnaround time for the travis tests.

**Pull request merged**